app/src/main/cpp/skyline/gpu.cpp

```diff
@@ -36,142 +37,38 @@ namespace skyline::gpu {
             surfaceUpdate = true;
             return;
         }
-        if (!bufferQueue.displayQueue.empty()) {
-            auto &buffer = bufferQueue.displayQueue.front();
-            bufferQueue.displayQueue.pop();
-            if (resolution != buffer->resolution || buffer->gbpBuffer.format != format) {
-                ANativeWindow_setBuffersGeometry(window, buffer->resolution.width, buffer->resolution.height, buffer->gbpBuffer.format);
-                resolution = buffer->resolution;
-                format = buffer->gbpBuffer.format;
+
+        if (!presentationQueue.empty()) {
+            auto &texture = presentationQueue.front();
+            presentationQueue.pop();
+
+            auto textureFormat = texture->GetAndroidFormat();
+            if (resolution != texture->dimensions || textureFormat != format) {
+                ANativeWindow_setBuffersGeometry(window, texture->dimensions.width, texture->dimensions.height, textureFormat);
+                resolution = texture->dimensions;
```

Comment:
May as well rename resolution to dimensions for consistency's sake

Reply:
I think there's somewhat of an implied meaning of resolution referring to display resolution on the other hand dimensions generally being used more in a texture/image settings. I've seen both being used in both but I think this is fine as is.

Reply:
hmm okay